app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -115,6 +118,21 @@ class GroupConversationDetailsViewModel @Inject constructor(
         updateGuestRemoteRequest(false)
     }
 
+    fun onServiceDialogDismiss() {
+        updateState(
+            groupOptionsState.copy(
+                loadingServicesOption = false,
+                changeServiceOptionConfirmationRequired = false,
+                isServicesAllowed = !groupOptionsState.isServicesAllowed
+            )
+        )
+    }
+
+    fun onServiceDialogConfirm() {
+        updateState(groupOptionsState.copy(changeServiceOptionConfirmationRequired = false, loadingServicesOption = true))
+        updateServicesRemoteRequest(false)
```

Comment:
why this is false?

Reply:
It's false because the dialog is to confirm disabling services
and the UI is updated before receiving the response to avoid the compose switch having an incorrect state.
You can check out this ticket
https://wearezeta.atlassian.net/browse/AR-2121